c++/src/kj/test.h

```diff
@@ -115,7 +116,7 @@ class TestCase {
 #else
 #define KJ_EXPECT_THROW_RECOVERABLE(type, code, ...) \
   do { \
-    KJ_IF_SOME(e, ::kj::runCatchingExceptions([&]() { code; })) { \
+    KJ_IF_SOME(e, ::kj::runCatchingExceptions([&]() { (void)({code;}); })) { \
```

Comment:
Oh that was trickier than I expected. But nice solution. Sorry that it made you chase MSVC issues.